decision-records/005-repository-organisation-for-v5.md

```diff
@@ -0,0 +1,120 @@
+
+This doesn't force us to ship the source files, as well, which we've decided to exclude (at least for now). Those can be excluded via an `.npmignore` files or the `files` field in `package.json`.
+
+### Removing built package files from versioning
+
+The review app uses relative paths (`../../src/...`) to access the components of GOV.UK Frontend. While we cannot make it consume `govuk-frontend` the exact same way our users do (downloading from an `npm` registry), we can get pretty close using workspaces. Making it use `govuk-frontend` directly helps ensure that:
+- we're consuming the built files, not rebuilding from source with a process that we have to keep in sync with how the package is built
+- the actual files work as intended
+- `entries` in `package.json` are properly configured
+- dependencies get resolved as expected
+
+For that to happen, we need the package rebuilt regularly rather than only at the point of release. If we were to leave it in the repository, there'd be a risk of committing built files that are in-between the previous release and the next one. Remove it it ensures we can rebuilt at will.
+
+### Updating our release process
+
+Versioning the built files serves as a way to check that there's no weird code sneaking in. Changes to the package get highlighted during the review of the pull request raised to commit the new build of the package and the release.
+
+We'll need to keep checking for unexplained changes to the package as we release, even if it is by manually diffing in a first time (before further automation).
+
+### Leaving `dist` in its current location
+
+`dist` doesn't need to move to enable the previous points, and with the amount of changes in v5, we'll leave it out of the equation for now.
+
+## Risks and constraints
+
+<!-- Any accepted risks associated with the decision. -->
+
+- There may be some confusion with `package` previously being somewhere not to edit the files of and now becoming where the sources are. Similarly, we'll have two `dist` folders (at root and in `package`). Naming might need updating to sort that out.
+
+## Alternatives considered
+
+<!-- Other options discussed but decided against. -->
+
+### Regrouping source and built package files within a shared package
+
+- Not regrouping the source and built package within the same package now. This would mean waiting for the next breaking change to do it. We may need to have peer dependencies sooner than it.
+
+### Removing built package files from versioning
+
+- We could also have built the package to a separate folder that's not versioned (for ex. `govuk-frontend-local-build`) to ensure we consume the built files without impacting `package`. 
```

Comment:
@colinrotherham @36degrees Do any of you remember why we left that one aside or was it just to avoid the clunkyness of building the same thing in two places with slightly different names?

Reply:
Yeah by replacing the real `govuk-frontend` in the workspace with the fake one we'd lose:

1. Dependency resolution testing
2. Package exports testing
3. Dependabot updates
4. ESLint import/export/publishing checks

Plus we'd end up replicating things in two `package.json` files

Probably more

Reply:
I'll update to try and reflect that 😄

Reply:
I was hoping the heading being under `## Alternatives considered` would clarify that it's the alternatives of `Removing built package` that are listed. It might be that `Regrouping source and built package...` only having one alternative makes things a little confusing.